### PR TITLE
Improve vrf test

### PR DESCRIPTION
**Pull request by tnasu**
## Description

* Refactor
* Take out and remake
  * crypto/vrf/compatibility_keys_test.go
  * crypto/vrf/compatibility_vrf_test.go
* Add hdevalence/ed25519consensus for test

### CODE REVIEW

File: crypto/vrf/compatibility_keys_test.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	coniks "github.com/coniks-sys/coniks-go/crypto/vrf"
+	ed25519consensus "github.com/hdevalence/ed25519consensus"
+	libsodium "github.com/line/ostracon/crypto/vrf/internal/vrf"
+	xed25519 "golang.org/x/crypto/ed25519"
+)
+
+var secret [SEEDBYTES]byte
+var message []byte = []byte("hello, world")
+
+func TestKeygen(t *testing.T) {
```
Comment (kokeshiM0chi) [inline crypto/vrf/compatibility_keys_test.go:23]: I am curious and have a question. If `curve25519-voi` is used in the future, will that test be added as well?

**Review by kokeshiM0chi — DISMISSED** [review]
LGTM

File: crypto/vrf/compatibility_keys_test.go (hunk 1 above)
Comment (tnasu) [reply crypto/vrf/compatibility_keys_test.go:23]: Yes, it does. I've already prepared that. But `curve25519-voi` needs `golang-1.16+`, so I've not put the commit these updates yet.

```diff
+       t.Run("ed25519 and voied25519 have compatibility",
+               func(t *testing.T) {
+                       testKeygen_compatibility(t, privateKey, publicKey,
+                               keygen_curve25519voi_ed25519, require.Equal, require.Equal)
+               })
```
```diff
+func keygen_curve25519voi_ed25519(secret [SEEDBYTES]byte) (ed25519.PrivateKey, ed25519.PublicKey) {
+       pubKey, privKey, _ := voied25519.GenerateKey(bytes.NewReader(secret[:]))
+       privateKey := make([]byte, voied25519.PrivateKeySize)
+       copy(privateKey, privKey[:])
+       publicKey := make([]byte, voied25519.PublicKeySize)
+       copy(publicKey, pubKey[:])
+       return privateKey, publicKey
+}
```
```diff
+func testKeypair_compatibility_curve25519voi_ed25519(
+       t *testing.T,
+       sk ed25519.PrivateKey, pk ed25519.PublicKey,
+       toPkTest func(t require.TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{}),
+       fromSkTest func(t require.TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{}),
+) {
+       pk1, sk1, _ := voied25519.GenerateKey(bytes.NewReader(secret[:]))
+
+       sk2 := ed25519.PrivateKey(make([]byte, ed25519.PrivateKeySize))
+       copy(sk2[:], sk1[:])
+       pk2 := sk2.Public().(ed25519.PublicKey)
+
+       toPkTest(t, pk1[:], pk2[:])
+
+       copy(sk1[:], sk[:])
+       pk1 = sk1.Public().(voied25519.PublicKey)
+
+       fromSkTest(t, pk1[:], pk2[:])
+}
```

And etc.

Comment (kokeshiM0chi) [reply crypto/vrf/compatibility_keys_test.go:23]: Oh, I see. Thank you for your explanation.

File: crypto/vrf/vrf_r2ishiguro_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+package vrf
+
+import (
+	"crypto/ed25519"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestProveAndVerify_r2ishiguro_by_crypto_ed25519(t *testing.T) {
```
Comment (torao) [inline crypto/vrf/vrf_r2ishiguro_test.go:13]: Why is this the only snake case?

**Review by torao — CHANGES_REQUESTED** [review]

File: crypto/vrf/vrf_r2ishiguro_test.go (hunk 2 above)
Comment (tnasu) [reply crypto/vrf/vrf_r2ishiguro_test.go:13]: I can only say that I was in the mood at that time.

Comment (tnasu) [reply crypto/vrf/vrf_r2ishiguro_test.go:13]: I feel that it's easy to look at the result on the display.

Comment (tnasu) [reply crypto/vrf/vrf_r2ishiguro_test.go:13]: Anyway, I'll change the function names with the camel case.

**Review by torao — APPROVED** [review]
LGTM

File: crypto/vrf/vrf_r2ishiguro_test.go (hunk 2 above)
Comment (torao) [reply crypto/vrf/vrf_r2ishiguro_test.go]: I also understand and agree that the snake case look is better in some cases, such as describing a combination of patterns, as in this case.

**Review by kokeshiM0chi — APPROVED** [review]

**Pull request merged**